transplant/transplant_remote.m

```diff
@@ -110,14 +110,29 @@ function transplant_remote(msgformat, url, zmqname, is_zombie)
                     if isKey(msg, 'nargout') && msg('nargout') >= 0
                         resultsize = msg('nargout');
                     else
+                        % nargout(fun) throws fails if fun is a method
+                        % nargout(msg('name')) can work on class methods, but
+                        % no way to specify class name in MATLAB x-(
+                        % do extra check on class name using which(msg('name'))
                         try
-                            resultsize = nargout(fun);
-                        catch % nargout fails if fun is a method:
-                            try
-                                resultsize = nargout(msg('name'));
-                            catch
-                                resultsize = -1;
+                            [~, funType] = which(msg('name'));
```

Comment:
In order to make this fast in the common case (functions), we should first use `isobject`/`isempty`) on the first argument and quickly fall back to `nargout` in the non-object case.